rust/ql/lib/codeql/rust/dataflow/internal/SsaImpl.qll

```diff
@@ -212,7 +212,8 @@ predicate capturedCallWrite(Expr call, BasicBlock bb, int i, Variable v) {
 /** Holds if `v` may be mutably borrowed in `e`. */
 private predicate mutablyBorrows(Expr e, Variable v) {
   e = any(MethodCallExpr mc).getReceiver() and
-  e.(VariableAccess).getVariable() = v
+  e.(VariableAccess).getVariable() = v and
+  v.isMutable()
```

Comment:
I think it would be nice to factor this bit out into a separate PR, with its own DCA run.

Reply:
https://github.com/github/codeql/pull/19024